app/org/sagebionetworks/bridge/models/schedules/Schedule.java

```diff
@@ -163,6 +165,14 @@ public boolean getPersistent() {
         }
         return false;
     }
+    @JsonIgnore
+    @DynamoDBIgnore
+    public TaskScheduler getScheduler() {
+        if (getCronTrigger() != null) {
+            return new CronTaskScheduler(this);
+        }
+        return new IntervalTaskScheduler(this);
+    }
```

Comment:
I feel like the ScheduleFactory was actually cleaner. ScheduleFactory provides a separation between data and behavior.